File: JS/DLL.js
/*
 * For your reference:
 *
 * DoublyLinkedListNode {
 *     int data;
 *     DoublyLinkedListNode next;
 *     DoublyLinkedListNode prev;
 * }
 *
 */

function sortedInsert(head, data) {
    const newNode = new DoublyLinkedListNode(data);
    // what if head is null
    if (head === null) {
      return newNode;
    }
    // if it should be inserted at the front, handle that case
    if (data <= head.data) {
      newNode.next = head;
      head.prev = newNode;
      return newNode;
    }
      // new.n = head
      // head.p = new
      // return new
    
    let current = head;
    while (current !== null) {
      if (current.data > data) {
        newNode.prev = current.prev;
        newNode.next = current;
        current.prev.next = newNode;
        current.prev = newNode;
        return head;
      }
      if (current.next == null) {
        break;
      }
      current = current.next;
    }
    
    current.next = newNode;
    newNode.prev = current;
    
    return head;
    // the list is sorted already
    // go through the list and check if data is < > the item in the LL
      // if data is > , then go to the next node
      // if data is < , then we insert it before our current node
        // insertion
  
    // if we get to the end without inserting our node, then we want to insert it at the end
  
    // edge cases
      // what if the data is less than the head?
      // what if we need to insert it at the very end of the LL
  }

File: Python/DLL.py
# DoublyLinkedListNode:
#     int data
#     DoublyLinkedListNode next
#     DoublyLinkedListNode prev
#
#


def sortedInsert(head, data):
  # create our newNode
  newNode = DoublyLinkedListNode(data)
  
  # what if head is null?
    # return newNode
  if head is None:
    return newNode
  
  # what if it needs to be inserted at the beginning?
  if head.data > newNode.data:
    # link the node onto the front of the list
    newNode.next = head
    head.prev = newNode
    # return newNode
    return newNode
    
  current = head
  # iterator over the LL,
  while current is not None:
    # check each node's data against the data we want to insert
    # when the node's data becomes greater than the toInsert data, 
    if current.data > newNode.data:
      # we want to insert the node before
      newNode.prev = current.prev
      newNode.next = current
      current.prev = newNode
      newNode.prev.next = newNode
      # return head
      return head
    
    
    # if we get to a point where next is null, then this means we need to insert it at the end
    if current.next is None:
      current.next = newNode
      newNode.prev = current
      return head
      # curr.next = newNode
      # newNode.prev = curr
      
    current = current.next
      
  
      
      
      
      
      
      
      
      
      
      
      
      
      
      
      